### ISSUE TITLE
How to merge with git at stormshiftdeploy?

### ISSUE THREAD

**Issue by mschindl** [stormshift/documentation#3]
How to merge with git at stormshiftdeploy?

Conflict with local files shown.
git pull > merge conflict

What is the common practice, because currently there is an inconsistency between the files?

**Comment by rbo** [comment]
At stormshiftdeploy?

**Comment by mschindl** [comment]
As discussed, we need a common workflow based on Git.

[root@stormshiftdeploy stormshift]# git status
On branch master
Your branch is behind 'origin/master' by 53 commits, and can be fast-forwarded.
[...]
>>> there are a lot of "Changes not staged for commit" and "Untracked files"

**Proposal**:
- Changes > Commit & Push immediately 
- Scheduled Forced-Pull every Sunday

**Comment by rbo** [comment]
> As discussed, we need a common workflow based on Git.
> 
> [root@stormshiftdeploy stormshift]# git status On branch master Your branch is behind 'origin/master' by 53 commits, and can be fast-forwarded. [...]
> 
> > > > there are a lot of "Changes not staged for commit" and "Untracked files"
> 
> **Proposal**:
> 
>     * Changes > Commit & Push immediately
Agree

> 
>     * Scheduled Forced-Pull every Sunday
Disagree, I suggest a force commit push from time to time. I don't like to lose changes. Or at least a monitoring & notification.

**Comment by mschindl** [comment]
> Changes > Commit & Push immediately

This is a "soft" rule and is ignored as we see it now.

>I suggest a force commit push from time to time

End of the month? A "hard" rule can help there.
If you don't commit, you won't forget again next time ;)

>I don't like to lose changes

Understandable and there we should find a way, especially with the growth of the Lab.

**Comment by DanielFroehlich** [comment]
so, the way I worked in that past is that I edited / changed the files locally on my laptop. If possible, I run playbooks from my laptop. Sometimes, doing this from the deploy-node is necessary. then I rsync'ed the files from my laptop to deploy-node.
This is ugly and bad practises. I just realized that the stormshift "checkout folder" on deploy-node is also from a wrong repo/branch.
I fixed that by moving it to "OLD_stormshift_20220713" folder and created a fresh  checkout from the master branch.
This should enable the "commit & push immideatly" approach.

In the long term, we should consider switching e.g. to personal login/users at the deploy-node, where multiple people can easily work concurrently.
